### DISCUSSION TITLE
option to not run userscripts in iframes and popups

### DISCUSSION THREAD

**Discussion by xBZZZZ** [violentmonkey/violentmonkey#1415] [Ideas]
can help avoid #1400 if enabled

**Comment by tophf** [comment]
I think we can fix #1400 completely. The fix I'm working on locally disables typical hacks. For more advanced hacks (not sure they would even work other than simply breaking Violentmonkey), I'll write a babel plugin that removes the ability to hack into the webpack's require system.

**Reply by tophf** [reply]
Note that such option can't be implemented anyway because there's no way for a WebExtension to detect that it runs in a hacked environment even in a normal top-level tab, which actually might have been created via window.open and spoofed by its opener.

**Comment by xBZZZZ** [comment]
nevermind it won't help because popup can abuse it's opener by reloading it
```javascript
open(`javascript:setTimeout(atob('${btoa(`
  //this is popup's script (runs after 1000 ms)
  opener.onunload=()=>setTimeout(()=>alert(document.cookie));//alerts cookie (you can see violentmonkey stuff if Synchronous mode is enabled)
  opener.location.reload();
`)}'),1000)`);
```
